tests/test_workers/test_pkg_managers/test_gomod.py

```diff
@@ -124,6 +125,122 @@ def test_resolve_gomod(
+        ),
+    ),
+)
+@mock.patch("cachito.workers.pkg_managers.gomod.get_golang_version")
+@mock.patch("cachito.workers.pkg_managers.gomod.GoCacheTemporaryDirectory")
+@mock.patch("subprocess.run")
+def test_resolve_gomod_vendor_dependencies(
+    mock_run,
+    mock_temp_dir,
+    mock_golang_version,
+    dep_replacement,
+    go_list_error_pkg,
+    expected_replace,
+    tmpdir,
+    sample_deps,
+    sample_deps_replace,
+    sample_deps_replace_new_name,
+    sample_package,
+):
+    mock_cmd_output = _generate_mock_cmd_output(go_list_error_pkg)
+    # mock_gwc.return_value.cachito_strict_gomod_vendor = True
+    mock_cmd_output = _generate_mock_cmd_output(go_list_error_pkg)
+    # Mock the tempfile.TemporaryDirectory context manager
+    mock_temp_dir.return_value.__enter__.return_value = str(tmpdir)
+
+    # Mock the "subprocess.run" calls
+    run_side_effects = []
+    if dep_replacement:
+        run_side_effects.append(mock.Mock(returncode=0, stdout=None))  # go mod edit -replace
+    run_side_effects.append(mock.Mock(returncode=0, stdout=None))  # go mod vendor
+    if dep_replacement:
+        run_side_effects.append(mock.Mock(returncode=0, stdout=None))  # go mod tidy
+    run_side_effects.append(mock.Mock(returncode=0, stdout=mock_cmd_output))  # go list -m all
+    mock_run.side_effect = run_side_effects
+
+    mock_golang_version.return_value = "v2.1.1"
+
+    archive_path = "/this/is/path/to/archive.tar.gz"
+    request = {
+        "id": 3,
+        "ref": "c50b93a32df1c9d700e3e80996845bc2e13be848",
+        "flags": ["gomod-vendor"],
+    }
+    if dep_replacement is None:
+        module, resolved_deps = resolve_gomod(archive_path, request)
+        expected_deps = sample_deps
+    else:
+        module, resolved_deps = resolve_gomod(archive_path, request, [dep_replacement])
+        if dep_replacement.get("new_name"):
+            expected_deps = sample_deps_replace_new_name
+        else:
+            expected_deps = sample_deps_replace
+
+    if expected_replace:
+        assert mock_run.call_args_list[0][0][0] == (
+            "go",
+            "mod",
+            "edit",
+            "-replace",
+            expected_replace,
+        )
+        if dep_replacement:
+            assert mock_run.call_args_list[2][0][0] == ("go", "mod", "tidy")
+
+    assert module == sample_package
+    assert resolved_deps == expected_deps
+
+
+@mock.patch("cachito.workers.pkg_managers.gomod.GoCacheTemporaryDirectory")
+@mock.patch("subprocess.run")
+@mock.patch("cachito.workers.pkg_managers.gomod.get_worker_config")
+@mock.patch("os.path.isdir")
+def test_resolve_gomod_strict_mode_raise_error(
+    mock_isdir, mock_gwc, mock_run, mock_temp_dir, tmpdir
+):
+    mock_isdir.return_value = True
+    # Mock the get_worker_config
+    celery_app = celery.Celery()
+    celery_app.cachito_strict_gomod_vendor = True
+    celery_app.cachito_athens_url = "http://athens:3000"
+    mock_gwc.return_value = celery_app
```

Comment:
You are setting these attributes on the Celery object and not its configuration. How about just using mock objects instead?
```suggestion
    mock_config = mock.Mock()
    mock_config.cachito_strict_gomod_vendor = True
    mock_config.cachito_athens_url = "http://athens:3000"
    mock_gwc.return_value = mock_config
```